File: btpy/on_work/structures/fit_list.js



/*
Function that adjusts the sent list
to the indicated size.
*/
export function fit_list(ARRAY, SIZE, 
        DEFAULT_VALUE = null) {
    //create copy of array
    let nuevoArray = [...ARRAY];
    // if is lower
    while (nuevoArray.length < SIZE) {
      nuevoArray.push(DEFAULT_VALUE);
    }
    // if is greater
    if (nuevoArray.length > SIZE) {
      nuevoArray.length = SIZE;
    }
    return nuevoArray;
}

File: btpy/src/btpy/btpy_list/mod/fit_list/fit_list.py



def fit_list(LIST:list, SIZE:int, 
        FILL_TYPE:any = None)->list:
    """
    Function that adjusts the sent list
    to the indicated size.
    """
    new_list = []
    for i in range(SIZE):
        if(i >= len(LIST)):
            new_list.append(FILL_TYPE)
        else:
            new_list.append(LIST[i])
    return new_list